cmd/kube-proxy/app/options/options.go

```diff
@@ -33,67 +34,59 @@ const (
-	Master                         string
-	Kubeconfig                     string
-	PortRange                      util.PortRange
-	HostnameOverride               string
-	ProxyMode                      string
-	IptablesSyncPeriod             time.Duration
-	ConfigSyncPeriod               time.Duration
-	NodeRef                        *api.ObjectReference // Reference to this node.
-	MasqueradeAll                  bool
-	CleanupAndExit                 bool
-	KubeAPIQPS                     float32
-	KubeAPIBurst                   int
-	UDPIdleTimeout                 time.Duration
-	ConntrackMax                   int
-	ConntrackTCPTimeoutEstablished int // seconds
+	componentconfig.KubeProxyConfiguration
+	ResourceContainer string
+	KubeAPIQPS        float32
+	KubeAPIBurst      int
+	ConfigSyncPeriod  time.Duration
+	CleanupAndExit    bool
+	NodeRef           *api.ObjectReference
+	Master            string
+	Kubeconfig        string
 }
 
 func NewProxyConfig() *ProxyServerConfig {
 	return &ProxyServerConfig{
-		BindAddress:                    net.ParseIP("0.0.0.0"),
-		HealthzPort:                    10249,
-		HealthzBindAddress:             net.ParseIP("127.0.0.1"),
-		OOMScoreAdj:                    qos.KubeProxyOOMScoreAdj,
-		ResourceContainer:              "/kube-proxy",
-		IptablesSyncPeriod:             30 * time.Second,
-		ConfigSyncPeriod:               15 * time.Minute,
-		KubeAPIQPS:                     5.0,
-		KubeAPIBurst:                   10,
-		UDPIdleTimeout:                 250 * time.Millisecond,
-		ConntrackMax:                   256 * 1024, // 4x default (64k)
-		ConntrackTCPTimeoutEstablished: 86400,      // 1 day (1/5 default)
+		KubeProxyConfiguration: componentconfig.KubeProxyConfiguration{
+			BindAddress:                    "0.0.0.0",
+			HealthzPort:                    10249,
+			HealthzBindAddress:             "127.0.0.1",
+			OOMScoreAdj:                    util.IntPtr(qos.KubeProxyOOMScoreAdj),
+			ResourceContainer:              "/kube-proxy",
+			IPTablesSyncPeriod:             unversioned.Duration{30 * time.Second},
+			UDPIdleTimeout:                 unversioned.Duration{250 * time.Millisecond},
+			Mode:                           componentconfig.ProxyModeUserspace,
+			ConntrackMax:                   256 * 1024,                                     // 4x default (64k)
+			ConntrackTCPEstablishedTimeout: unversioned.Duration{Duration: 24 * time.Hour}, // 1 day (1/5 default)
+		},
+		KubeAPIQPS:       5.0,
+		KubeAPIBurst:     10,
+		ConfigSyncPeriod: 15 * time.Minute,
 	}
 }
 
 // AddFlags adds flags for a specific ProxyServer to the specified FlagSet
 func (s *ProxyServerConfig) AddFlags(fs *pflag.FlagSet) {
-	fs.IPVar(&s.BindAddress, "bind-address", s.BindAddress, "The IP address for the proxy server to serve on (set to 0.0.0.0 for all interfaces)")
+	fs.StringVar(&s.BindAddress, "bind-address", s.BindAddress, "The IP address for the proxy server to serve on (set to 0.0.0.0 for all interfaces)")
```

Comment:
I tried to do this but Tim said it was a bad idea. https://github.com/kubernetes/kubernetes/pull/18689#issuecomment-164644315